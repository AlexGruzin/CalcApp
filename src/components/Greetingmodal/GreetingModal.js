import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/Button';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import './GreetingModal.styles.less';

const GreetingModal = ({ className }) => {
  const { t } = useTranslation();

  return (
    <div className={classNames('greet-ml', className)}>
      <h1 className="greet-ml__hey">Hey There</h1>
      <p className="greet-ml__description">Welcome to the club!</p>
      <Button className="greet-ml__tour" onClick={() => {}}>
        {t('Take a site tour')}
      </Button>
      <Button className="greet-ml__store" onClick={() => {}}>
        {t('Take Store Survey')}
      </Button>
    </div>
  );
};

GreetingModal.propTypes = {
  className: PropTypes.string,
};

GreetingModal.defaultProps = {
  className: '',
};

export default GreetingModal;
