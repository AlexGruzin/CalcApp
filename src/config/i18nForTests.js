import i18n from 'i18next';

// locales
import US from '../locales/en';

i18n.init({
  debug: false,
  fallbackLng: ['US'],
  lng: 'US',

  ns: ['auth', 'resetPassword'],
  defaultNS: 'auth',

  resources: {
    US,
  },
});

export default i18n;

