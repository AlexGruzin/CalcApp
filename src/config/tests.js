import React from 'react';
import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next'
import i18n from "./i18nForTests";
import jsdom from 'jsdom';
const { JSDOM } = jsdom;

/*
 * jest-dom
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom/extend-expect';

/*
 * jest-fetch-mock
 * https://github.com/jefflau/jest-fetch-mock
 */
require('jest-fetch-mock').enableMocks();

if (global.document) {
  global.document.createRange = () => ({
    setStart: () => {
    },
    setEnd: () => {
    },
    commonAncestorContainer: {
      nodeName: 'BODY',
      ownerDocument: document,
    },
    createContextualFragment: str => JSDOM.fragment(str),
    // createContextualFragment: str => document.createElement('hr'),
  });
  global.open = () => undefined;
}

export const renderWithi18next = Component => {
  const Comp = React.cloneElement(Component, {
    changeLanguage: lng => {
      i18n.changeLanguage(lng);
      rerender(<I18nextProvider i18n={i18n}>{Comp}</I18nextProvider>);
    }
  });
  const defaultRender = render(
    <I18nextProvider i18n={i18n}>{Component}</I18nextProvider>
  );
  const { rerender } = defaultRender;
  return defaultRender;
};


// /**
//  * Wraps a component with an Intl Provider.
//  * @param {Object} component The component to render
//  */
// const withTestIntl = component => <IntlProvider locale="en">{component}</IntlProvider>;
//
// /*
//  * https://testing-library.com/docs/example-react-intl
//  */
// export const renderWithReactIntl = component => render(withTestIntl(component));

