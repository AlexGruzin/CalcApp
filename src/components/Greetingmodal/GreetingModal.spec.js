import React from 'react';

import { renderWithi18next } from 'config/tests';

import GreetingModal from './GreetingModal';

describe('GreetingModal tests', () => {
  it('Check snapshots of GreetingModal are equal.', () => {
    const { asFragment } = renderWithi18next(<GreetingModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
