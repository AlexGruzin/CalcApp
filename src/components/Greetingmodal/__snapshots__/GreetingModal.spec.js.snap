// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreetingModal tests Check snapshots of GreetingModal are equal. 1`] = `
<DocumentFragment>
  <div
    class="greet-ml"
  >
    <h1
      class="greet-ml__hey"
    >
      Hey There
    </h1>
    <p
      class="greet-ml__description"
    >
      Welcome to the club!
    </p>
    <button
      class="custom-button greet-ml__tour"
      type="button"
      value=""
    >
      Take a site tour
    </button>
    <button
      class="custom-button greet-ml__store"
      type="button"
      value=""
    >
      Take Store Survey
    </button>
  </div>
</DocumentFragment>
`;
